File: problems/3396-minimum-number-of-operations-to-make-elements-in-array-distinct/solution.py
class Solution:
    def minimumOperations(self, nums: List[int]) -> int:
        n = len(nums)
        if n == 0:
            return 0

        for ops in range((n + 2) // 3 + 1):
            remaining = nums[ops * 3:]
            if len(set(remaining)) == len(remaining):
                return ops
            if len(remaining) == 0:
                return ops

File: problems/3396-minimum-number-of-operations-to-make-elements-in-array-distinct/solution.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int minimumOperations(int[] nums) {
        int n = nums.length;
        if (n == 0) {
            return 0;
        }
        int operations = 0;
        for (int i = 0; i <= n / 3; i++) {
            int remainingLength = n - i * 3;
            if (remainingLength <= 0) {
                return i;
            }
            int[] remainingNums = new int[remainingLength];
            for (int j = 0; j < remainingLength; j++) {
                remainingNums[j] = nums[i * 3 + j];
            }
            Set<Integer> distinctElements = new HashSet<>();
            boolean distinct = true;
            for (int num : remainingNums) {
                if (distinctElements.contains(num)) {
                    distinct = false;
                    break;
                }
                distinctElements.add(num);
            }
            if (distinct) {
                return i;
            }
        }
        return n / 3 + 1;
    }
}